src/main/java/com/pimp/domain/Calendar.java

```diff
@@ -78,4 +79,16 @@ public boolean isPrivate() {
   public void setPrivate(boolean aPrivate) {
     isPrivate = aPrivate;
   }
+
+  public Event getEventByKey(String eventKey) {
+    Optional<Event> firstEvent = getEvents()
+      .stream()
+      .filter(event -> event.getKey().equals(eventKey))
+      .findFirst();
+    return firstEvent.isPresent() ? firstEvent.get() : null;
+  }
+
+  public boolean containsEvent(String eventKey) {
```

Comment:
Wouldn't it be much better to override `equals()` [true when two events have the same `key`]] and `hashCode()` in the `Events` class (IntelliJ can generate this)? Then you can just check `calendar.getEvents().contains(event)` and you can get the index of an event in the List of Events with `indexOf(event)`.